dashboard/modules/state/state_head.py

```diff
@@ -339,6 +339,30 @@ async def list_runtime_envs(self, req: aiohttp.web.Request) -> aiohttp.web.Respo
         record_extra_usage_tag(TagKey.CORE_STATE_API_LIST_RUNTIME_ENVS, "1")
         return await self._handle_list_api(self._state_api.list_runtime_envs, req)
 
+    def filter_events(self, events, severity_levels, source_types, **params):
+        # If custom field parameters are provided, extract them
+        entity_name, entity_id = list(params.items())[0] if params else (None, None)
```

Comment:
This code is confusing. Does it assume the first query always includes entity_name, entity_id? Can we just explicitly pass it to the filter method?

Reply:
That is a good idea. entity_name and entity_id is optional in the HTTP request